p2p/netutil/addrutil_test.go

```diff
@@ -0,0 +1,191 @@
+package netutil
+
+import (
+	"math/rand"
+	"net"
+	"net/netip"
+	"path/filepath"
+	"testing"
+)
+
+// customNetAddr is a custom implementation of net.Addr for testing purposes.
+type customNetAddr struct{}
+
+func (c *customNetAddr) Network() string { return "custom" }
+func (c *customNetAddr) String() string  { return "custom" }
+
+func TestAddrAddr(t *testing.T) {
+	tempDir := t.TempDir()
+	tests := []struct {
+		name string
+		addr net.Addr
+		want netip.Addr
+	}{
+		{
+			name: "IPAddr IPv4",
+			addr: &net.IPAddr{IP: net.ParseIP("192.0.2.1")},
+			want: netip.MustParseAddr("192.0.2.1"),
+		},
+		{
+			name: "IPAddr IPv6",
+			addr: &net.IPAddr{IP: net.ParseIP("2001:db8::1")},
+			want: netip.MustParseAddr("2001:db8::1"),
+		},
+		{
+			name: "TCPAddr IPv4",
+			addr: &net.TCPAddr{IP: net.ParseIP("192.0.2.1"), Port: 8080},
+			want: netip.MustParseAddr("192.0.2.1"),
+		},
+		{
+			name: "TCPAddr IPv6",
+			addr: &net.TCPAddr{IP: net.ParseIP("2001:db8::1"), Port: 8080},
+			want: netip.MustParseAddr("2001:db8::1"),
+		},
+		{
+			name: "UDPAddr IPv4",
+			addr: &net.UDPAddr{IP: net.ParseIP("192.0.2.1"), Port: 8080},
+			want: netip.MustParseAddr("192.0.2.1"),
+		},
+		{
+			name: "UDPAddr IPv6",
+			addr: &net.UDPAddr{IP: net.ParseIP("2001:db8::1"), Port: 8080},
+			want: netip.MustParseAddr("2001:db8::1"),
+		},
+		{
+			name: "Unsupported Addr type",
+			addr: &net.UnixAddr{Name: filepath.Join(tempDir, "test.sock"), Net: "unix"},
+			want: netip.Addr{},
+		},
+		{
+			name: "Nil input",
+			addr: nil,
+			want: netip.Addr{},
+		},
+		{
+			name: "Custom net.Addr implementation",
+			addr: &customNetAddr{},
+			want: netip.Addr{},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := AddrAddr(tt.addr); got != tt.want {
+				t.Errorf("AddrAddr() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestIPToAddr(t *testing.T) {
+	tests := []struct {
+		name string
+		ip   net.IP
+		want netip.Addr
+	}{
+		{
+			name: "IPv4",
+			ip:   net.ParseIP("192.0.2.1"),
+			want: netip.MustParseAddr("192.0.2.1"),
+		},
+		{
+			name: "IPv6",
+			ip:   net.ParseIP("2001:db8::1"),
+			want: netip.MustParseAddr("2001:db8::1"),
+		},
+		{
+			name: "Invalid IP",
+			ip:   net.IP{1, 2, 3},
+			want: netip.Addr{},
+		},
+		{
+			name: "Invalid IP (5 octets)",
+			ip:   net.IP{192, 0, 2, 1, 1},
+			want: netip.Addr{},
+		},
+		{
+			name: "IPv4-mapped IPv6",
+			ip:   net.ParseIP("::ffff:192.0.2.1"),
+			want: netip.MustParseAddr("192.0.2.1"),
+		},
+		{
+			name: "Nil input",
+			ip:   nil,
+			want: netip.Addr{},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := IPToAddr(tt.ip); got != tt.want {
+				t.Errorf("IPToAddr() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestRandomAddr(t *testing.T) {
```

Comment:
I think this is a bit overkill. The `RandomAddr` is ~15 LOC, and essentially just generates random bytes which is translated to `ipv4/ipv6`. It is very trivial, and it is also _only_ invoked from tests. So adding a test for it is very meta. Let's just ensure it works, once, then never bother with it again.  So IMO drop this method

Reply:
`TestRandomAddr` has been removed